hazelcast-client/src/main/java/com/hazelcast/client/map/impl/nearcache/ClientHeapNearCache.java

```diff
@@ -260,4 +196,71 @@ public NearCacheStatsImpl getNearCacheStats() {
         stats.setOwnedEntryMemoryCost(ownedEntryMemoryCost);
         return stats;
     }
+
+    private void fireCacheEviction() {
```

Comment:
I would pull out this anonymous innerclass into a regular inner class. The anoymous inner class quickly lead to painful to read code, but also cause problems when analyzing heapdumps since you don't get to see a propper class name.